low-level/files/opt/seagate/sspl/setup/resource_map/server.py

```diff
@@ -32,18 +39,163 @@ class ServerMap(ResourceMap):
 
         rpath: Resource id (Example: nodes[0]>compute[0]>hw>disks)
         """
-        pass
+        info = {}
+        fru_found = False
+        nodes = rpath.strip().split(">")
+        leaf_node, _ = self.get_node_details(nodes[-1])
+        if leaf_node == "compute":
+            for fru in self.server_frus:
+                info.update({fru: self.server_frus[fru]()})
+            info["last_updated"] = int(time.time())
+            fru_found = True
+        else:
+            fru = None
+            for node in nodes:
+                fru, _ = self.get_node_details(node)
+                if self.server_frus.get(fru):
+                    fru_found = True
+                    info = self.server_frus[fru]()
+                    break
+        if not fru_found:
+            raise ResourceMapError(
+                errno.EINVAL,
+                "Health provider doesn't have support for'{rpath}'.")
+        return info
+
+    @staticmethod
+    def get_node_details(node):
+        """
+        Parse node information and returns left string and instance.
+
+        Example
+            "storage"    -> ("storage", "*")
+            "storage[0]" -> ("storage", "0")
+        """
+        res = re.search(r"(\w+)\[([\d]+)\]|(\w+)", node)
+        inst = res.groups()[1] if res.groups()[1] else "*"
+        node = res.groups()[0] if res.groups()[1] else res.groups()[2]
+        return node, inst
+
+    @classmethod
+    def get_sysfs_base_path(cls):
+        """Returns the sysfs base path. Ex: /sys."""
+        sysfs_base_path = Conf.get(SSPL_CONF,
+                                   "SYSTEM_INFORMATION>sysfs_base_path", '/sys/')
+        return sysfs_base_path
+
+    @staticmethod
+    def get_data_template(uid, is_fru: bool):
+        """Returns health template."""
+        return {
+            "uid": uid,
+            "fru": str(is_fru).lower(),
+            "last_updated": int(time.time()),
+            "health": {
+                "status": "",
+                "description": "",
+                "recommendation": "",
+                "specifics": []
+            }
+        }
+
+    @staticmethod
+    def set_health_data(health_data: dict, status, description=None,
+                        recommendation=None, specification=None):
+        """Sets health attributes for a component."""
+        good_state = (status == "OK")
+        if not description:
+            description = "%s %s in good health" % (
+                health_data.get("uid"),
+                'is' if good_state else 'is not')
+        if not recommendation:
+            recommendation = 'None' if good_state\
+                             else "Fault detected, please contact Seagate support."
+
+        health_data["health"].update({
+            "status": status,
+            "description": description,
+            "recommendation": recommendation,
+            "specifics": specification
+        })
+
+    @staticmethod
+    def convert_cpu_info_list(cpu_info):
```

Comment:
Hi @jjxsg,  all these methods `convert_cpu_info_list`, `get_sysfs_base_path`, `get_cpu_info` are present in `framework/utils/sysfs_interface.py`. We can use the methods available there or modify those methods according to the requirement. 
I have tried the the same in PR https://github.com/Seagate/cortx-monitor/pull/414/files, please check if applicable.

Reply:
@sumedhak27 Thanks for the suggestion. Earlier I was planning to make changes in `convert_cpu_info_list` fuction, thats why I kept it within the file itself. But if the functions are unchanged, using them directly from `framework/utils/sysfs_interface.py` makes more sense. 
I have implemented the above changes you suggested and got rid of `convert_cpu_info_list `and  `get_sysfs_base_path` from `server.py`. 
`get_cpu_info` is a modified function, so I will need to keep it here.